File: app/src/main/java/com/qwert2603/vkmessagestat/di/PresenterModule.java
package com.qwert2603.vkmessagestat.di;

import com.qwert2603.vkmessagestat.model.DataManager;
import com.qwert2603.vkmessagestat.model.DataManagerImpl;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class PresenterModule {

    @Provides
    @Singleton
    DataManager mDataManager() {
        return new DataManagerImpl();
    }

}

File: app/src/main/java/com/qwert2603/vkmessagestat/di/PresenterModule.kt
package com.qwert2603.vkmessagestat.di

import com.qwert2603.vkmessagestat.model.DataManager
import com.qwert2603.vkmessagestat.model.DataManagerImpl
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class PresenterModule {

    @Provides
    @Singleton
    internal fun mDataManager(): DataManager {
        return DataManagerImpl()
    }

}
